misk-redis/src/main/kotlin/misk/redis/FakeRedis.kt

```diff
@@ -25,96 +27,126 @@ class FakeRedis : Redis {
     ConcurrentHashMap<String, Value<ConcurrentHashMap<String, ByteString>>>()
 
   override fun del(key: String): Boolean {
-    if (!keyValueStore.containsKey(key)) {
-      return false
-    }
+    synchronized(lock) {
```

Comment:
I added this lock usage because it seemed important to avoid any concurrency errors